guides/theming.md

```diff
@@ -1,311 +1,359 @@
-particular, a color configuration consists of:
+This document describes the concepts and APIs for customizing colors. For typography customization,
+see [Angular Material Typograpy][mat-typography]. For guidance on building components to be
+customizable with this system, see [Theming your own components][theme-your-own].
 
-* A **primary** palette: colors most widely used across all screens and components.
-* An **accent** palette: colors used for the floating action button and interactive elements.
-* A **warn** palette: colors used to convey error state.
-* A **foreground** palette: colors for text and icons.
-* A **background** palette: colors used for element backgrounds.
+[material-design-theming]: https://material.io/design/material-theming/overview.html
+[mat-typography]: https://material.angular.io/guide/typography
+[theme-your-own]: https://material.angular.io/guide/theming-your-components
 
-Additionally, in Angular Material, a configuration may optionally include `typography` settings.
-More information on how typography works can be [found in a dedicated guide][3].
+### Sass
 
-Angular Material theme styles are generated _statically_ at build-time so that your
-app doesn't have to spend cycles generating theme styles when bootstrapping.
+Angular Material's theming APIs are built with [Sass](https://sass-lang.com). This document assumes
+familiary with CSS and Sass basics, including variables, functions, and mixins.
 
-## Using a pre-built theme
+You can use Angular Material without Sass by using a pre-built theme, described in
+[Using a pre-built theme](#using-a-pre-built-theme) below. However, using the library's Sass API
+directly gives you the most control over the styles in your application.
 
-Angular Material comes prepackaged with several pre-built theme css files. These theme files also
-include all the styles for the core (styles common to all components), so you only have to include a
-single css file for Angular Material in your app.
+## Palettes
 
-You can include a theme file directly into your application from
-`@angular/material/prebuilt-themes`
+A **palette** is a collection of colors representing a portion of color space. Each value in this
+collection is called a **hue**. In Material Design, the hues in a palette are numbered from zero
+to 900, lightest to darkest.
 
-Available pre-built themes:
-* `deeppurple-amber.css`
-* `indigo-pink.css`
-* `pink-bluegrey.css`
-* `purple-green.css`
+Angular Material represents a palette as a [Sass map][sass-maps]. This map contains the
+palette's hues and another nested map of contrast colors for each of the hues. The contrast colors
+serve as text color when using a hue as a background color. The example below demonstrates the
+structure of a palette.
 
-If you're using Angular CLI, this is as simple as including one line
-in your `styles.css`  file:
-```css
-@import '@angular/material/prebuilt-themes/deeppurple-amber.css';
+```scss
+$indigo: (
```

Comment:
This was renamed to `$indigo-palette`.

Reply:
This was marked resolved but still named `$indigo`